rust_src/src/filelock.rs

```diff
@@ -0,0 +1,46 @@
+//! Lock files for editing.
+
+use remacs_macros::lisp_fn;
+
+use crate::{
+    lisp::LispObject,
+    remacs_sys::{lock_file, unlock_file},
+    remacs_sys::Qstringp,
```

Comment:
Thanks. We do this in an effort to cut down on git conflicts as we add new entries.